pandas/io/excel/_base.py

```diff
@@ -450,7 +457,22 @@ def parse(
             else:  # assume an integer if not a string
                 sheet = self.get_sheet_by_index(asheetname)
 
-            data = self.get_sheet_data(sheet, convert_float)
+            if isinstance(header, int):
+                header_nrows = header
+            elif header is None:
+                header_nrows = 0
+            else:
+                header_nrows = max(header)
+            if isinstance(skiprows, int):
+                skiprows_nrows = skiprows
+            elif skiprows is None:
+                skiprows_nrows = 0
+            else:
+                skiprows_nrows = len(skiprows)
```

Comment:
this looks like its causing failures on master

Reply:
yep: https://dev.azure.com/pandas-dev/pandas/_build/results?buildId=43000&view=logs&j=b1c7b65e-b3ce-541a-7fd5-29b4ba56ce18&t=46ecc253-e38f-5abc-2ea7-addca6b44d0a

@jbrockmendel can you revert for now and we can reopen